File: opendemo_output/nodejs/nodejs-nodejs-class-inheritance-demo/code/employee.js
/**
 * employee.js - 员工管理系统类示例
 * 展示属性扩展与方法重写
 */

class Employee {
  constructor(name, position) {
    this.name = name;
    this.position = position;
  }

  getDetails() {
    return `员工姓名: ${this.name}，职位: ${this.position}`;
  }
}

// Manager 继承 Employee 并扩展功能
class Manager extends Employee {
  constructor(name, department, teamSize) {
    // 必须首先调用 super() 来初始化父类部分
    super(name, '经理');
    this.department = department;
    this.teamSize = teamSize;
  }

  // 重写方法以包含更多细节
  getDetails() {
    return `💼 经理姓名: ${this.name}，部门: ${this.department}，团队人数: ${this.teamSize}`;
  }
}

module.exports = { Employee, Manager };

File: opendemo_output/go/go-go-demo-12/employee.go
package main

import "fmt"

// Employee 表示员工，通过组合Person扩展功能
type Employee struct {
	Person  // 匿名嵌入Person，自动获得其字段和方法
	Position string // 职位
}

// Display 显示员工完整信息
func (e Employee) Display() {
	fmt.Printf("员工: %s, 职位: %s, 年龄: %d\n", e.Name, e.Position, e.Age)
}